simulator/infra/macro.h

```diff
@@ -177,6 +177,13 @@ static constexpr T ones_ls(const T& value, size_t shamt)
     return ~(~value << shamt);
 }
 
+template<typename T>
+static constexpr T rotate_l(const T& value, size_t shamt)
```

Comment:
Please follow the style of other shifting functions (rotate_ls).
Also, please put a space after bracket to follow our code guidelines.